src/js/components/Drop/Drop.js

```diff
@@ -1,50 +1,50 @@
-import React, { Component } from 'react';
+import React, { useEffect } from 'react';
 import { createPortal } from 'react-dom';
 
 import { getNewContainer, setFocusWithoutScroll } from '../../utils';
 
 import { DropContainer } from './DropContainer';
 
-class Drop extends Component {
-  static defaultProps = {
-    align: {
-      top: 'top',
-      left: 'left',
-    },
-    plain: false,
-  };
-
-  originalFocusedElement = document.activeElement;
-
-  dropContainer = getNewContainer();
-
-  componentWillUnmount() {
-    const { restrictFocus } = this.props;
-    if (restrictFocus && this.originalFocusedElement) {
-      if (this.originalFocusedElement.focus) {
-        setFocusWithoutScroll(this.originalFocusedElement);
-      } else if (
-        this.originalFocusedElement.parentNode &&
-        this.originalFocusedElement.parentNode.focus
-      ) {
-        // required for IE11 and Edge
-        setFocusWithoutScroll(this.originalFocusedElement.parentNode);
+const Drop = props => {
+  const originalFocusedElement = document.activeElement;
+
+  const dropContainer = getNewContainer();
+
+  useEffect(() => {
+    return () => {
+      const { restrictFocus } = props;
+      if (restrictFocus && originalFocusedElement) {
+        if (originalFocusedElement.focus) {
+          setFocusWithoutScroll(originalFocusedElement);
+        } else if (
+          originalFocusedElement.parentNode &&
+          originalFocusedElement.parentNode.focus
+        ) {
+          // required for IE11 and Edge
+          setFocusWithoutScroll(originalFocusedElement.parentNode);
+        }
       }
-    }
-    document.body.removeChild(this.dropContainer);
-  }
-
-  render() {
-    const {
-      target: dropTarget, // avoid DOM leakage
-      ...rest
-    } = this.props;
-    return createPortal(
-      <DropContainer dropTarget={dropTarget} {...rest} />,
-      this.dropContainer,
-    );
-  }
-}
+      document.body.removeChild(dropContainer);
+    };
+  });
+
+  const {
+    target: dropTarget, // avoid DOM leakage
+    ...rest
+  } = props;
+  return createPortal(
+    <DropContainer dropTarget={dropTarget} {...rest} />,
+    dropContainer,
+  );
+};
+
+Drop.defaultProps = {
+  align: {
+    top: 'top',
```

Comment:
Looking into the align props I was thinking to create some constants eg:
`TOP_LEFT, TOP_RIGHT, BOTTOM_RIGHT`
Create a map with all the possible combinations and document this options.

Reply:
This will be nice but i think it could be a breaking change. What do you think @ShimiSun ?

Reply:
I like the idea, I'm thinking it will be cleaner and maybe help with compiling every string separately by compiling it just once. Also, it might help with avoiding human spelling mistakes :) I'd say let's give it a try on a separate PR.